pkg/clusteragent/admission/controllers/webhook/common.go

```diff
@@ -8,13 +8,59 @@
 package webhook
 
 import (
+	"encoding/json"
+	agentsidecar "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate/agent_sidecar"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common"
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate"
 	"github.com/DataDog/datadog-agent/pkg/config"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Selector specifies an object label selector and a namespace label selector
+type Selector struct {
+	ObjectSelector    metav1.LabelSelector `yaml:"objectSelector,omitempty"`
+	NamespaceSelector metav1.LabelSelector `yaml:"namespaceSelector,omitempty"`
+}
+
+// buildAgentSidecarObjectSelectors returns the mutating webhooks object selectors based on the configuration
+func buildAgentSidecarObjectSelectors() (namespaceSelector, objectSelector *metav1.LabelSelector) {
```

Comment:
This is specific for the agent sidecar, shouldn't it be defined somewhere in the new `agent_sidecar` package instead of this `common` file ?

Reply:
I definitely agree with you 👍 

However, I am following the existing pattern.

For example, `buildCWSInstrumentationLabelSelectors` is specific to CWS Instrumentation, yet it is defined in this `common.go` file.

I thought it would be good to stick with the existing pattern until the admission controller code is refactored to follow a structured organisation.

WDYT?

Reply:
We can leave it for later :+1:

That would be part of the refactor I'm working on now.